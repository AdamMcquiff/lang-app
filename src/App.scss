.l-app {
  display: flex;

  &__header {
    flex: 1;
  }

  &__main {
    flex: 6;
    padding: 1em;
  }
}