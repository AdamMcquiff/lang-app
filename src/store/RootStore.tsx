import SettingsService from '../services/settingsService/SettingsService';
import TranslationCategoryService from '../services/translationCategoriesService/TranslationCategoryService';
import TranslationService from '../services/translationService/TranslationService';
import SettingsStore from './SettingsStore';
import TranslationCategoryStore from './TranslationCategoryStore';
import TranslationStore from './TranslationStore';

class RootStore {
	public settingsStore: SettingsStore;
	public translationCategoryStore: TranslationCategoryStore;
	public translationStore: TranslationStore;

	constructor() {
		this.settingsStore = new SettingsStore(this, SettingsService);
		this.translationCategoryStore = new TranslationCategoryStore(this, TranslationCategoryService);
		this.translationStore = new TranslationStore(this, TranslationService);
	}
}

export default RootStore;