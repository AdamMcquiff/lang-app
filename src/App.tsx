import { observer } from 'mobx-react';
import * as React from 'react';
import './App.scss';
import TranslationsLayout from './patterns/layouts/translations/TranslationsLayout';
import { Header } from './patterns/organisms/header/Header';
import RootStore from './store/RootStore';

interface IAppInterface {
    store: RootStore
}

@observer
class App extends React.Component<IAppInterface> {
    public componentDidMount() { 
        this._requestApiDataFromServices();
    }

    public render() {
        return (
            <div className="l-app">
                <Header className="l-app__header" 
                        store={this.props.store} />
                        
                <TranslationsLayout className="l-app__main" 
                                    store={this.props.store} />
            </div>
        );
    }

    private _requestApiDataFromServices() {
        this.props.store.settingsStore.list();
        this.props.store.translationCategoryStore.list();
        this.props.store.translationStore.list();
    }
}

export default App;
