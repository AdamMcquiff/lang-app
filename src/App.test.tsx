import { shallow } from 'enzyme';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import App from './App';
import RootStore from './store/RootStore';

const store: RootStore = new RootStore();

it('renders without crashing', () => {
    const component = shallow(<App store={store} />);
    expect(component).toMatchSnapshot();
});