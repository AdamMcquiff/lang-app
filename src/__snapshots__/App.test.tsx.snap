// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="l-app"
>
  <Header
    className="l-app__header"
    store={
      RootStore {
        "settingsStore": SettingsStore {
          "rootStore": [Circular],
          "settings": Object {
            "native_lang": "",
            "translated_lang": "",
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "SettingsStore@2.settings.keys",
                "observers": Set {},
              },
              "name": "SettingsStore@2.settings",
              "pendingKeys": Map {
                "cheerio" => false,
                "$$typeof" => false,
                "nodeType" => false,
                "@@__IMMUTABLE_ITERABLE__@@" => false,
                "@@__IMMUTABLE_RECORD__@@" => false,
                "_isMockFunction" => false,
                "toJSON" => false,
              },
              "proxy": [Circular],
              "target": Object {
                "native_lang": "",
                "translated_lang": "",
                Symbol(mobx administration): [Circular],
              },
              "values": Map {
                "native_lang" => "",
                "translated_lang" => "",
              },
            },
          },
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "SettingsStore@2.keys",
              "observers": Set {},
            },
            "name": "SettingsStore@2",
            "target": [Circular],
            "values": Map {
              "settings" => Object {
                "native_lang": "",
                "translated_lang": "",
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "SettingsStore@2.settings.keys",
                    "observers": Set {},
                  },
                  "name": "SettingsStore@2.settings",
                  "pendingKeys": Map {
                    "cheerio" => false,
                    "$$typeof" => false,
                    "nodeType" => false,
                    "@@__IMMUTABLE_ITERABLE__@@" => false,
                    "@@__IMMUTABLE_RECORD__@@" => false,
                    "_isMockFunction" => false,
                    "toJSON" => false,
                  },
                  "proxy": [Circular],
                  "target": Object {
                    "native_lang": "",
                    "translated_lang": "",
                    Symbol(mobx administration): [Circular],
                  },
                  "values": Map {
                    "native_lang" => "",
                    "translated_lang" => "",
                  },
                },
              },
            },
          },
        },
        "translationStore": TranslationStore {
          "createTranslation": [Function],
          "createTranslationCategory": [Function],
          "deleteTranslation": [Function],
          "getTranslationCategories": [Function],
          "getTranslations": [Function],
          "rootStore": [Circular],
          "translationCategories": Array [],
          "translations": Array [],
          "updateTranslation": [Function],
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "TranslationStore@1.keys",
              "observers": Set {},
            },
            "name": "TranslationStore@1",
            "target": [Circular],
            "values": Map {
              "translationCategories" => Array [],
              "translations" => Array [],
            },
          },
        },
      }
    }
  />
  <main
    className="l-app__main"
  />
</div>
`;
